services/the-giving-blocks/syncProjectsCronJob.ts

```diff
@@ -0,0 +1,183 @@
+      givingBlocksCategory,
+    });
+  }
+};
+
+const createGivingProject = async (data: {
+  accessToken: string;
+  givingBlockProject: GivingBlockProject;
+  givingBlocksCategory: GivingBlocksCategory;
+}) => {
+  const { accessToken, givingBlockProject, givingBlocksCategory } = data;
+  try {
+    if (givingBlockProject.allowsAnon === false) return;
+
+    const givethProject = await Project.findOne({
+      givingBlocksId: String(givingBlockProject.id),
+    });
+    console.log(
+      `GivingBlocksProject ${
+        givingBlockProject.id
+      }. Exists: ${!!givethProject}`,
+    );
+
+    if (!givethProject) {
+      const walletAddress = await generateGivingBlockDepositAddress(
+        accessToken,
+        givingBlockProject.id,
+      );
+
+      const organization = await fetchOrganizationById(
+        accessToken,
+        givingBlockProject.id,
+      );
+
+      // Await enough for full limit to regenerate
+      await sleep(1000);
+
+      const description = organization?.websiteBlocks?.missionStatement?.value;
+      const youtube = organization?.websiteBlocks?.youtubeUrl?.value;
+      const website = organization?.websiteBlocks?.url?.value;
+
+      // Our Current Validations
+      const slugBase = slugify(givingBlockProject.name, {
+        remove: /[*+~.,()'"!:@]/g,
+      });
+      const slug = await getAppropriateSlug(slugBase);
+      const qualityScore = getQualityScore(
+        description,
+        Boolean(givingBlockProject.logo),
+      );
+
+      const project = Project.create({
+        title: givingBlockProject.name,
+        description,
+        categories: [givingBlocksCategory],
+        walletAddress,
+        creationDate: new Date(),
+        slug,
+        youtube,
+        website,
+        slugHistory: [],
+        givingBlocksId: String(givingBlockProject.id),
+        admin: adminId,
+        statusId: ProjStatus.active,
+        qualityScore,
+        totalDonations: 0,
+        totalReactions: 0,
+        totalProjectUpdates: 0,
+        listed: true,
+        verified: true,
+        giveBacks: true,
+      });
+
+      await project.save();
+
+      // create default projectUpdate to allow adding Reactions
+      const update = ProjectUpdate.create({
+        userId: Number(adminId),
+        projectId: project.id,
+        content: '',
+        title: '',
+        createdAt: new Date(),
+        isMain: true,
+      });
+
+      await ProjectUpdate.save(update);
+    }
+  } catch (e) {
+    // Log Error but keep going with the rest of the projects
+    console.log(e);
+  }
+};
+
+// Current Formula: This will be changed in the future
+const getQualityScore = (description, hasImageUpload) => {
+  let qualityScore = 40;
+
+  // Some projects have no description
+  if (Number(description?.length) > 100) qualityScore = qualityScore + 10;
+  if (hasImageUpload) qualityScore = qualityScore + 30;
+
+  return qualityScore;
+};
+
+const getAppropriateSlug = async (slugBase: string) => {
```

Comment:
@CarlosQ96  please specify output type for this function
PS: We used that before, that time we didn't write type, but it's good to write types from now on